File: Leetcode 3901-4000/3941/3941.java
// Leetcode 3941: Password Strength
// https://leetcode.com/problems/password-strength/
// Solved on 7th of June, 2026
class Solution {
    /**
     * Calculates the strength of a password based on unique character types.
     * @param password The input string to evaluate.
     * @return The total calculated strength score.
     */
    public int passwordStrength(String password) {
        boolean[] seen = new boolean[128];
        int totalStrength = 0;
        for (int i = 0; i < password.length(); i++) {
            char ch = password.charAt(i);
            if (!seen[ch]) {
                seen[ch] = true;
                if (ch >= 'a' && ch <= 'z') {
                    totalStrength += 1;
                } else if (ch >= 'A' && ch <= 'Z') {
                    totalStrength += 2;
                } else if (ch >= '0' && ch <= '9') {
                    totalStrength += 3;
                } else if (ch == '!' || ch == '@' || ch == '#' || ch == '$') {
                    totalStrength += 5;
                }
            }
        }
        return totalStrength;
    }
}

File: Leetcode 3901-4000/3941/3941.py
# Leetcode 3941:  Password Strength
# https://leetcode.com/problems/password-strength/
# Solved on 7th of June, 2026
class Solution:
    def passwordStrength(self, password: str) -> int:
        """
        Calculates the strength of a password based on the types of unique characters it contains.

        :param password: The input string representing the password to evaluate.
        :return: An integer representing the total calculated strength.
        """
        uniqueChars = set(password)
        totalStrength = 0

        for eachChar in uniqueChars:
            if 'a' <= eachChar <= 'z':
                totalStrength += 1
            elif 'A' <= eachChar <= 'Z':
                totalStrength += 2
            elif '0' <= eachChar <= '9':
                totalStrength += 3
            elif eachChar in "!@#$":
                totalStrength += 5

        return totalStrength